cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -136,29 +142,24 @@ public void destroy() {
     // Nothing to do
   }
 
-  private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
+  private int tryParse(String val) {
```

Comment:
we're catching a NumberFormatException, returning -1, and then detecting the -1 and rethrowing an exception.
I think it's better just to not catch the NFE in the first place.